backend/tests/test_crud_schema.py

```diff
@@ -364,288 +313,280 @@ def test_update(self, dbsession):
-        bfk = dbsession.execute(
-            select(BoundFK)
-            .where(BoundFK.schema_id == 1)
-            .where(BoundFK.attr_def_id == test_def.id)
-        ).scalar()
-        assert bfk
-
-    def test_raise_on_reserved_slig(self, dbsession):
-        for i in RESERVED_SCHEMA_SLUGS:
-            upd_schema = SchemaUpdateSchema(
-                name='Test', 
-                slug=i, 
-                update_attributes=[], 
-                add_attributes=[]
-            )
-            with pytest.raises(ReservedSchemaSlugException):
-                update_schema(dbsession, id_or_slug=1, data=upd_schema)
+        assert nickname.attribute.type == AttrType.DT
+        assert all([nickname.required, nickname.list, nickname.key])
 
-    def test_raise_on_schema_doesnt_exist(self, dbsession):
+    def test_raise_on_renaming_to_already_present_attr(self, dbsession):
         upd_schema = SchemaUpdateSchema(
             name='Test', 
             slug='test', 
-            update_attributes=[], 
-            add_attributes=[]
+            update_attributes=[
+                AttrDefUpdateSchema(
+                    name='nickname',
+                    new_name='friends',
+                    required=False,
+                    unique=False,
+                    list=False,
+                    key=False,
+                    description='updated'
+                )
+            ], 
+            add_attributes=[],
+            delete_attributes=[]
         )
-        with pytest.raises(MissingSchemaException):
-            update_schema(dbsession, id_or_slug=99999999, data=upd_schema)
-
-    def test_raise_on_existing_slug_or_name(self, dbsession):
-        new_sch = Schema(name='Test', slug='test')
-        dbsession.add(new_sch)
-        dbsession.flush()
-        
-        upd_schema = SchemaUpdateSchema(name='Person', slug='test', update_attributes=[], add_attributes=[])
-        with pytest.raises(SchemaExistsException):
-            update_schema(dbsession, id_or_slug=new_sch.id, data=upd_schema)
+        with pytest.raises(MultipleAttributeOccurencesException):
+            update_schema(dbsession, id_or_slug=1, data=upd_schema)
         dbsession.rollback()
-        dbsession.add(new_sch)
-        dbsession.flush()
-
-        upd_schema = SchemaUpdateSchema(name='Test', slug='person', update_attributes=[], add_attributes=[])
-        with pytest.raises(SchemaExistsException):
-            update_schema(dbsession, id_or_slug=new_sch.id, data=upd_schema)
 
-    def test_raise_on_attr_def_doesnt_exist(self, dbsession):
+        upd_schema.update_attributes = [
+             AttrDefUpdateSchema(
+                    name='nickname',
+                    new_name='fav_color',
+                    required=False,
+                    unique=False,
+                    list=False,
+                    key=False,
+                    description='updated'
+                )
+        ]
```

Comment:
The purpose of this statement eludes me. Did you intended to test more?

Reply:
Oh, actually that was for testing with updated data, but when I updated API, this test was no longer necessary, but I forgot to remove data for it😅